embassy-rp/src/watchdog.rs

```diff
@@ -0,0 +1,111 @@
+//! Watchdog
+//!
+//! The watchdog is a countdown timer that can restart parts of the chip if it reaches zero. This can be used to restart the
+//! processor if software gets stuck in an infinite loop. The programmer must periodically write a value to the watchdog to
+//! stop it from reaching zero.
+//!
+//! Credit: based on `rp-hal` implementation (also licensed Apache+MIT)
+
+use core::marker::PhantomData;
+
+use embassy_time::Duration;
+
+use crate::pac;
+use crate::peripherals::WATCHDOG;
+
+/// Watchdog peripheral
+pub struct Watchdog<'d> {
```

Comment:
the `'d` lifetime is not doing anything because `new` takes full ownership of WATCHDOG, could you remove it?